site/src/pages/WorkspacePage/WorkspaceNotifications.tsx

```diff
@@ -0,0 +1,343 @@
+      title: "Workspace using deprecated template",
+      severity: "warning",
+      detail: template.deprecation_message,
+    });
+  }
+
+  const infoNotifications = notifications.filter((n) => n.severity === "info");
+  const warningNotifications = notifications.filter(
+    (n) => n.severity === "warning",
+  );
+
+  return (
+    <div
+      css={{
+        display: "flex",
+        alignItems: "center",
+        gap: 12,
+        position: "fixed",
+        bottom: 48,
+        right: 48,
+        zIndex: 10,
+      }}
+    >
+      {infoNotifications.length > 0 && (
+        <NotificationPill
+          notifications={infoNotifications}
+          type="info"
+          icon={<InfoOutlined />}
+        />
+      )}
+
+      {warningNotifications.length > 0 && (
+        <NotificationPill
+          notifications={warningNotifications}
+          type="warning"
+          icon={<WarningRounded />}
+        />
+      )}
+    </div>
+  );
+};
+
+type NotificationPillProps = {
+  notifications: Notification[];
+  type: ThemeRole;
+  icon: ReactNode;
+};
+
+const NotificationPill: FC<NotificationPillProps> = (props) => {
+  const { notifications, type, icon } = props;
+  const theme = useTheme();
+
+  return (
+    <Popover mode="hover">
+      <PopoverTrigger>
+        <div css={[styles.pillContainer]}>
+          <Pill type={type} icon={icon}>
+            {notifications.length}
+          </Pill>
+        </div>
+      </PopoverTrigger>
+      <PopoverContent
+        transformOrigin={{
+          horizontal: "right",
+          vertical: "bottom",
+        }}
+        anchorOrigin={{ horizontal: "right", vertical: "top" }}
+        css={{
+          "& .MuiPaper-root": {
+            borderColor: theme.experimental.roles[type].outline,
+            maxWidth: 400,
+          },
+        }}
+      >
+        {notifications.map((n) => (
+          <NotificationItem notification={n} key={n.title} />
+        ))}
+      </PopoverContent>
+    </Popover>
+  );
+};
+
+const NotificationItem: FC<{ notification: Notification }> = (props) => {
+  const { notification } = props;
+  const theme = useTheme();
+
+  return (
+    <article css={{ padding: 16 }}>
+      <h4 css={{ margin: 0, fontWeight: 500 }}>{notification.title}</h4>
+      {notification.detail && (
+        <p
+          css={{
+            margin: 0,
+            color: theme.palette.text.secondary,
+            lineHeight: 1.6,
+          }}
+        >
+          {notification.detail}
+        </p>
+      )}
+      <div css={{ marginTop: 8 }}>{notification.actions}</div>
+    </article>
+  );
+};
+
+const NotificationActionButton: FC<ButtonProps> = (props) => {
+  return (
+    <Button
+      variant="text"
+      css={{
+        textDecoration: "underline",
+        padding: 0,
+        height: "auto",
+        minWidth: "auto",
+        "&:hover": { background: "none", textDecoration: "underline" },
+      }}
+      {...props}
+    />
+  );
+};
+
+const styles = {
```

Comment:
there's even a `styles` object here to use! :)

Reply:
🤣